src/org/intellij/grammar/parser/GeneratedParserUtilBase.java

```diff
@@ -720,14 +719,25 @@ else if ((frame.modifiers & _LEFT_) != 0 && frame.leftMarker != null) {
     else if (result || pinned) {
       if (marker != null) marker.drop();
       if ((frame.modifiers & _LEFT_INNER_) != 0 && frame.leftMarker != null) {
-        frame.leftMarker.precede().done(((LighterASTNode)frame.leftMarker).getTokenType());
-        frame.leftMarker.drop();
+        closeLeft(frame);
       }
     }
     else {
       close_marker_impl_(frame, marker, null, false);
     }
   }
+  
+  private static void closeLeft(Frame frame) {
+    LighterASTNode leftMarker = (LighterASTNode) frame.leftMarker;
+    PsiBuilder.Marker precede = frame.leftMarker.precede();
+    if (leftMarker.getTokenType() == TokenType.ERROR_ELEMENT) {
+      precede.error("");
```

Comment:
why not `PsiBuilderImpl.getErrorMessage(leftMarker)` instead ?

Reply:
OK, right, but it's nullable, so
```
      String message = PsiBuilderImpl.getErrorMessage(leftMarker);
      precede.error(StringUtil.notNullize(message));
```